### PR TITLE
Fixed issues with URL traveling

### PR DESCRIPTION
**Pull request by Vatyx**
#### Description
Two issues
* `+useExternalIpForBridge` was not respected even if there were no params regarding this option sent in the URL.
* When traveling, and the client map differed from the server, URL options were dropped.

#### Primary reviewers
@m-samiec @joshuahuburn

### CODE REVIEW

File: SpatialGDK/Source/Private/EngineClasses/SpatialNetDriver.cpp [hunk 1]
```diff
@@ -153,7 +153,18 @@ void USpatialNetDriver::OnMapLoaded(UWorld* LoadedWorld)
 			Connection->ReceptionistConfig.ReceptionistPort = LoadedWorld->URL.Port;
 		}
 
-		Connection->ReceptionistConfig.UseExternalIp = LoadedWorld->URL.HasOption(TEXT("useExternalIpForBridge"));
+		FString UseExternalIpOption = FString(LoadedWorld->URL.GetOption(TEXT("useExternalIpForBridge"), TEXT("")));
+		if(!UseExternalIpOption.IsEmpty())
+		{
+			if (UseExternalIpOption.ToLower().Equals(TEXT("true")))
```
Comment (m-samiec) [inline SpatialGDK/Source/Private/EngineClasses/SpatialNetDriver.cpp]:
```suggestion
			if (UseExternalIpOption.Equals(TEXT("true"), ESearchCase::IgnoreCase))
```

File: SpatialGDK/Source/Private/EngineClasses/SpatialNetDriver.cpp [hunk 2]
```diff
@@ -153,7 +153,18 @@ void USpatialNetDriver::OnMapLoaded(UWorld* LoadedWorld)
...
+			{
+				Connection->ReceptionistConfig.UseExternalIp = true;
+			}
+			else if (UseExternalIpOption.ToLower().Equals(TEXT("false")))
```
Comment (m-samiec) [inline SpatialGDK/Source/Private/EngineClasses/SpatialNetDriver.cpp]:
```suggestion
			else if (UseExternalIpOption.Equals(TEXT("false"), ESearchCase::IgnoreCase))
```

File: SpatialGDK/Source/Private/EngineClasses/SpatialNetDriver.cpp [hunk 3]
```diff
@@ -277,10 +288,9 @@ void USpatialNetDriver::OnAcceptingPlayersChanged(bool bAcceptingPlayers)
 			// Extract map name and options
 			FWorldContext* WorldContext = GEngine->GetWorldContextFromWorld(GetWorld());
 			check(WorldContext);
-			
-			FURL RedirectURL = FURL(&WorldContext->LastURL, *GlobalStateManager->DeploymentMapURL, (ETravelType)WorldContext->TravelType);
-			RedirectURL.Host = WorldContext->LastURL.Host;
-			RedirectURL.Port = WorldContext->LastURL.Port;
+
+			FURL RedirectURL = WorldContext->LastURL;
+			RedirectURL.Map = GlobalStateManager->DeploymentMapURL;
```
Comment (m-samiec) [inline SpatialGDK/Source/Private/EngineClasses/SpatialNetDriver.cpp]: If the MapURL is passed in via the constructor, FURL does a ton of processing on it that we lose by assigning direct to Map.

File: SpatialGDK/Source/Private/EngineClasses/SpatialNetDriver.cpp [hunk 4]
```diff
@@ -153,7 +153,18 @@ void USpatialNetDriver::OnMapLoaded(UWorld* LoadedWorld)
 			Connection->ReceptionistConfig.ReceptionistPort = LoadedWorld->URL.Port;
 		}
 
-		Connection->ReceptionistConfig.UseExternalIp = LoadedWorld->URL.HasOption(TEXT("useExternalIpForBridge"));
+		FString UseExternalIpOption = FString(LoadedWorld->URL.GetOption(TEXT("useExternalIpForBridge"), TEXT("")));
```
Comment (joshuahuburn) [inline SpatialGDK/Source/Private/EngineClasses/SpatialNetDriver.cpp]: This is a boolean option no? So this should already be correct. 
The existence of the option itself indicates the boolean value.
`"/Maps/MyMap?useExternalIpForBridge"` would set `ReceptionistConfig.UseExternalIp = true`
`"/Maps/MyMap"` would set `ReceptionistConfig.UseExternalIp = false`

Don't think this change is needed.

File: SpatialGDK/Source/Private/EngineClasses/SpatialNetDriver.cpp [hunk 5]
```diff
@@ -277,10 +288,9 @@ void USpatialNetDriver::OnAcceptingPlayersChanged(bool bAcceptingPlayers)
 			// Extract map name and options
 			FWorldContext* WorldContext = GEngine->GetWorldContextFromWorld(GetWorld());
 			check(WorldContext);
-			
-			FURL RedirectURL = FURL(&WorldContext->LastURL, *GlobalStateManager->DeploymentMapURL, (ETravelType)WorldContext->TravelType);
```
Comment (joshuahuburn) [inline SpatialGDK/Source/Private/EngineClasses/SpatialNetDriver.cpp]: The constructor explicitly parses the options of the second parameter `FURL(&WorldContext->LastURL, *GlobalStateManager->DeploymentMapURL, (ETravelType)WorldContext->TravelType);` in this case the `*GlobalStateManager->DeploymentMapURL`.
If you skip the constructor I believe the options will not be respected and thus you'll lose the options specified in the `DeploymentMapURL`

I think the problem is options that were used in the client travel being dropped when connecting to a server with a different map? 
If so we should update the `DeploymentMapURL` to have the correct options.

Comment (joshuahuburn) [reply SpatialGDK/Source/Private/EngineClasses/SpatialNetDriver.cpp]: Add a comment to indicate this constructor is to keep the URL options.

**Pull request merged**